osu.Game/Screens/Play/SkipOverlay.cs

```diff
@@ -143,24 +143,35 @@ protected override void LoadComplete()
         /// </summary>
         public void SkipWhenReady()
         {
-            if (skipQueued) return;
+            if (SkipQueued) return;
 
-            skipQueued = true;
+            SkipQueued = true;
             attemptNextSkip();
 
             void attemptNextSkip() => Scheduler.AddDelayed(() =>
             {
-                if (!button.Enabled.Value)
+                if (!isClickable || !gameplayClock.IsRunning)
```

Comment:
What is this change doing? Isn't `isClickable` the same as `button.Enabled.Value`, looking at `Update()`? And what is the condition with the clock doing here?

Reply:
`isClickable` is supposed to be the same. I'll move clock check to `Update()`.

Reply:
The reply above does not answer my question. I didn't ask to move the clock check anywhere, I asked what it was _doing_. As in, what is its _purpose_. Because _it didn't exist before_.

It's insanely difficult to review pull requests that seem to have essentially random changes applied in them back and forth without a clear end goal or explanation.

Reply:
> There were cases previously where clock was not running

_Which_ cases, specifically?

Reply:
Whether skipping is applied before of after `StartGameplay` depends on the combination of 250ms timer of `SkipWhenReady` and thread synchronization. That is, if my assumption that `load` and `StartGameplay` are called on different threads which i havent checked, but i am fairly confident in assuming.